### DISCUSSION TITLE
Add time to log.

### DISCUSSION THREAD

**Discussion by ozgurerdogan** [psy0rz/zfs_autobackup#233] [General]
How can I add full date to logs. My cron is:
sh /Backup-Vms.sh > /Backup-Vms.log 2>&1

and is sh:
ssh boxtobackup "zfs-autobackup -v  s1d1 D8 --keep-source=40 --keep-target=40 --strip-path=2 --clear-mountpoint --clear-refreservation --no-holds --snapshot-format %Y-%m-%d-%H:%M:%S --force --destroy-missing=10d";

And logs are:
#### Synchronising
  [Source] D2/vm-138-disk-0: sending to D8/vm-138-disk-0
  [Source] D2/vm-138-disk-0@2023-11-06-17:00:03: Destroying
  [Target] D8/vm-138-disk-0@2023-11-06-17:00:03: Destroying
  [Target] D8/vm-138-disk-0@2023-11-12-11:00:03: receiving incremental
  [Source] D2/vm-179-disk-0: sending to D8/vm-179-disk-0

So can I prepend date and time to logs?

**Comment by psy0rz** [comment]
you want to prepend the date/time to every line?

**Reply by ozgurerdogan** [reply]
yes

**Comment by psy0rz** [comment]
Pipe the output through this:
```
(your command...)| awk '{ print strftime("[%Y-%m-%d %H:%M:%S]"), $0; fflush(); }'
```

yes, gpt4 did this :)

**Reply by ozgurerdogan** [reply]
So something like this:

(ssh boxtobackup "zfs-autobackup -v s1d1 D8 --keep-source=40 --keep-target=40 --strip-path=2 --clear-mountpoint --clear-refreservation --no-holds --snapshot-format %Y-%m-%d-%H:%M:%S --force --destroy-missing=10d";) | awk '{ print strftime("[%Y-%m-%d %H:%M:%S]"), $0; fflush(); }'
